test/components/guideLineLayerTests.ts

```diff
@@ -0,0 +1,401 @@
+      assert.strictEqual((<any>linearScale)._callbacks.size, 0, "callback was removed from the previous Scale");
+    });
+
+    it("changing the scale updates value() if pixelPosition() is set but value() is not", () => {
+      let gll = new Plottable.Components.GuideLineLayer<number>("vertical");
+      let position = 50;
+      gll.pixelPosition(position);
+
+      let linearScale = new Plottable.Scales.Linear();
+      linearScale.domain([0, 1]);
+      linearScale.range([0, 100]);
+
+      gll.scale(linearScale);
+      let expectedValue = linearScale.invert(position);
+      assert.strictEqual(gll.value(), expectedValue, "value was updated to match the set pixel position");
+    });
+  });
+
+  it("rejects invalid orientations", () => {
+    assert.doesNotThrow(() => new Plottable.Components.GuideLineLayer<number>("vertical"), Error, "accepts \"vertical\"");
+    assert.doesNotThrow(() => new Plottable.Components.GuideLineLayer<number>("horizontal"), Error, "accepts \"horizontal\"");
+    assert.throws(() => new Plottable.Components.GuideLineLayer<number>("blargh"), Error);
+  });
+
+  describe("rendering (vertical)", () => {
+    let SVG_WIDTH = 400;
+    let SVG_HEIGHT = 300;
+
+    it("has an effective size of 0, but will occupy all offered space", () => {
+      let gll = new Plottable.Components.GuideLineLayer<void>("vertical");
+      let request = gll.requestedSpace(SVG_WIDTH, SVG_HEIGHT);
+      TestMethods.verifySpaceRequest(request, 0, 0, "does not request any space");
+      assert.isTrue(gll.fixedWidth(), "fixed width");
+      assert.isTrue(gll.fixedHeight(), "fixed height");
+
+      let svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
+      gll.anchor(svg);
+      gll.computeLayout({x: 0, y: 0}, SVG_WIDTH, SVG_HEIGHT);
+      assert.strictEqual(gll.width(), SVG_WIDTH, "accepted all offered width");
+      assert.strictEqual(gll.height(), SVG_HEIGHT, "accepted all offered height");
+      svg.remove();
+    });
+
+    it("clipPath enabled", () => {
+      let svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
+      let gll = new Plottable.Components.GuideLineLayer<void>("vertical");
+      gll.renderTo(svg);
+      TestMethods.verifyClipPath(gll);
+      let clipRect = (<any> gll)._boxContainer.select(".clip-rect");
+      assert.strictEqual(TestMethods.numAttr(clipRect, "width"), SVG_WIDTH, "the clipRect has an appropriate width");
+      assert.strictEqual(TestMethods.numAttr(clipRect, "height"), SVG_HEIGHT, "the clipRect has an appropriate height");
+      svg.remove();
+    });
+
+    it("renders correctly given a pixel position", () => {
+      let svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
+      let expectedPosition1 = SVG_WIDTH / 2;
+      let gll = new Plottable.Components.GuideLineLayer<number>("vertical");
+      gll.pixelPosition(expectedPosition1);
+      gll.renderTo(svg);
+
+      assert.strictEqual(gll.content().selectAll("line").size(), 1, "exactly one line is drawn");
```

Comment:
For this and the rest of the tests, can we use the more specific ".guide-line" identifier?